lib/Twig/TokenParser/AutoEscape.php

```diff
@@ -39,7 +39,7 @@ public function parse(Twig_Token $token)
         } else {
             $expr = $this->parser->getExpressionParser()->parseExpression();
             if (!$expr instanceof Twig_Node_Expression_Constant) {
-                throw new Twig_Error_Syntax('An escaping strategy must be a string or a bool.', $stream->getCurrent()->getLine(), $stream->getFilename());
+                throw new Twig_Error_Syntax('An escaping strategy must be a string or a bool.', $stream->getCurrent()->getLine(), $this->parser->getFilename());
```

Comment:
why not using ``$stream->getSourceContext()->getName()`` directly to avoid 1 level in the stack each time ? the parser will forward the call to the stream anyway